titapapunne/MoreThanWords
# Mtw 65 Template homescreen pod component, fetch data with query string & Reuse code in backend

**HomeScreenPod.js**

- contain template for Home Screen pod component to account for future changes to avoid code repeat.
- fetch data using the **GET** method sending the query _pod=${pod}_ to the backend

**app.py**

- Extract pod from the request arguments from frontend
- reuse code to get data in sf for each pod
- send back only progress and total outcomes for the current pod

> //  previous changes(reverted)
> - Added to the outcomes endpoint in app.py to calculate the total number of outcomes and then used that info in the frontend of HomeScreen.js for the progress bars; should still work if new outcomes are added
> - Use desc and "Outcomes__c" to filter related field
> - transform json data to python dictionary(easier to process data without hardcoding) 
> - process data to get the sum for all outcomes in the Trainee pod


**HomeScreen.js**

- create an array that contains all the pod names
- render the three pods on the home screen by passing data to HomeScreenPod 

> //  previous changes(reverted)
> - Change to using real user data instead of fake user data by passing in the user parameter in app.py and by using token in HomeScreen.js (also changed from axios to fetch).

---------------------------------------------------------------------------------------

Steps to test PR:
Login in on app using Jumbo1's account to test having real user data:

Username: titapa.punpun@gmail.com      
password: JumboOne!!

Then 
- the Home Screen should load all the progress bars with the correct number of total outcomes
- The first pod(Trainee) should be highlighted
- Navigation to the Trainee, Associate, and Partner all work fine

> Note:
> Css highlight logic: when _complete_ is smaller than _total_ (ongoing) and is _not zero_(have started)
> -  it won't work for some of the current mock data, but it will work in real user scenario -- when user are not allowed to start next pod (i.e have complete outcome with the new pod) unless they finish the previous one.

---------------------------------------------------------------------------------------

Standard Checklist:
